includes/class-content.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Republication Tracker Tool Content.
+ *
+ * @since   1.0
+ * @package Republication_Tracker_Tool
+ */
+
+/**
+ * Republication Tracker Tool Content class.
+ *
+ * @since 1.0
+ */
+class Republication_Tracker_Tool_Content {
+	/**
+	 * Filter the content for the republication.
+	 *
+	 * @param string $content The post content.
+	 * @param int    $post_id The post ID – should be supplied if different than the current post ID.
+	 */
+	public static function get_republishable_content( $content, $post_id = false ) {
+		if ( ! $post_id ) {
+			$post_id = get_the_ID();
+		}
+
+		// Remove shortcodes from the content.
+		$content = strip_shortcodes( $content );
+
+		// Remove comments from the content. (Lookin' at you, Gutenberg.)
+		$content = preg_replace( '/<!--(.|\s)*?-->/i', ' ', $content );
+
+		// And finally, remove some tags.
+		$content = wp_kses( $content, $allowed_tags_excerpt );
```

Comment:
I might be missing something but `$allowed_tags_excerpt` doesn't seem to be defined anywhere?

Reply:
Good catch! Updated